cin_validator/cin_validator_class.py

```diff
@@ -10,18 +14,42 @@
 
 class CinValidationSession:
     def __init__(self, filename, ruleset, issue_id=None) -> None:
-        # TODO detect filetype xml/csv/zip. check if the directory is a folder.
-        fulltree = ET.parse(filename)
-        root = fulltree.getroot()
-        self.data_files_obj = DataContainerWrapper(XMLtoCSV(root))
-
         self.ruleset = ruleset
         self.issue_id = issue_id
+        self.filename = filename
 
+        self.file_type_checker()
         self.create_error_report_df()
         self.create_json_report()
         self.select_by_id()
 
+    def file_type_checker(self):
+        fn = self.filename.name
+        if os.path.isfile(fn):
+            extension = pathlib.Path(fn).suffix
+            if extension[:4] == ".xml":
+                fulltree = ET.parse(fn)
+                root = fulltree.getroot()
+                self.data_files_obj = DataContainerWrapper(XMLtoCSV(root))
+                print("got here!")
+            elif extension[:4] == ".csv":
+                print(".csv support coming soon!")
```

Comment:
If the file is a csv, I wonder if we are able to read the filename (or if it is safe to assume that the file names will be what we expect them to be).
A reliable means of processing csv files will probably be to read their column names and deduce the name of the file (table). For example, a file that contains CPPstartDate is probably the ChildProtectionPlans table. 
Coding this up might be too verbose but first let's make it work and then we make it better.

I understand that we have listed out the columns for each table in two places already (api.py and ingress.py). It won't be nice to do that a third time. Let's figure that out.